pkg/controller/activemqartemis/activemqartemis_reconciler.go

```diff
@@ -228,6 +228,11 @@ func sourceEnvVarFromSecret(customResource *brokerv2alpha1.ActiveMQArtemis, curr
 		// Check the contents against what we just got above
 		elem, ok := nettySecret.Data[envVarName]
 		if 0 != strings.Compare(string(elem), acceptorEntry) || !ok {
+
+			if nettySecret.Data == nil {
+				nettySecret.Data = map[string][]byte{}
```

Comment:
Did you test this scenario? Not following the whole code, but OTTOMH, you might need make()

Reply:
reverted this change, I was getting nil panic/error  'assignment to entry in nil map' at this line but that might be due to my StatefulSet or other objects not created with the fake client,